Add sendSMS connector files: Swagger, apiProperties, and README

ElenaMainerici opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

ElenaMainerici committed:
Add sendSMS connector files: Swagger, apiProperties, and README

ElenaMainerici commented:
@microsoft-github-policy-service agree company="sendSMS"

On Sat, May 11, 2024 at 9:57 AM microsoft-github-policy-service[bot] <
***@***.***> wrote:

> @ElenaMainerici <https://github.com/ElenaMainerici> please read the
> following Contributor License Agreement(CLA). If you agree with the CLA,
> please reply with the following information.
>
> @microsoft-github-policy-service agree [company="{your company}"]
>
> Options:
>
>    - (default - no company specified) I have sole ownership of
>    intellectual property rights to my Submissions and I am not making
>    Submissions in the course of work for my employer.
>
> @microsoft-github-policy-service agree
>
>
>    - (when company given) I am making Submissions in the course of work
>    for my employer (or my employer has intellectual property rights in my
>    Submissions by contract or applicable law). I have permission from my
>    employer to make Submissions and enter into this Agreement on behalf of my
>    employer. By signing below, the defined term “You” includes me and my
>    employer.
>
> @microsoft-github-policy-service agree company="Microsoft"
>
> Contributor License Agreement Contribution License Agreement
>
> This Contribution License Agreement (*“Agreement”*) is agreed to by the
> party signing below (*“You”*),
> and conveys certain license rights to Microsoft Corporation and its
> affiliates (“Microsoft”) for Your
> contributions to Microsoft open source projects. This Agreement is
> effective as of the latest signature
> date below.
>
>    1. *Definitions*.
>    *“Code”* means the computer software code, whether in human-readable
>    or machine-executable form,
>    that is delivered by You to Microsoft under this Agreement.
>    *“Project”* means any of the projects owned or managed by Microsoft
>    and offered under a license
>    approved by the Open Source Initiative (www.opensource.org).
>    *“Submit”* is the act of uploading, submitting, transmitting, or
>    distributing code or other content to any
>    Project, including but not limited to communication on electronic
>    mailing lists, source code control
>    systems, and issue tracking systems that are managed by, or on behalf
>    of, the Project for the purpose of
>    discussing and improving that Project, but excluding communication
>    that is conspicuously marked or
>    otherwise designated in writing by You as “Not a Submission.”
>    *“Submission”* means the Code and any other copyrightable material
>    Submitted by You, including any
>    associated comments and documentation.
>    2. *Your Submission*. You must agree to the terms of this Agreement
>    before making a Submission to any
>    Project. This Agreement covers any and all Submissions that You, now
>    or in the future (except as
>    described in Section 4 below), Submit to any Project.
>    3. *Originality of Work*. You represent that each of Your Submissions
>    is entirely Your original work.
>    Should You wish to Submit materials that are not Your original work,
>    You may Submit them separately
>    to the Project if You (a) retain all copyright and license information
>    that was in the materials as You
>    received them, (b) in the description accompanying Your Submission,
>    include the phrase “Submission
>    containing materials of a third party:” followed by the names of the
>    third party and any licenses or other
>    restrictions of which You are aware, and (c) follow any other
>    instructions in the Project’s written
>    guidelines concerning Submissions.
>    4. *Your Employer*. References to “employer” in this Agreement include
>    Your employer or anyone else
>    for whom You are acting in making Your Submission, e.g. as a
>    contractor, vendor, or agent. If Your
>    Submission is made in the course of Your work for an employer or Your
>    employer has intellectual
>    property rights in Your Submission by contract or applicable law, You
>    must secure permission from Your
>    employer to make the Submission before signing this Agreement. In that
>    case, the term “You” in this
>    Agreement will refer to You and the employer collectively. If You
>    change employers in the future and
>    desire to Submit additional Submissions for the new employer, then You
>    agree to sign a new Agreement
>    and secure permission from the new employer before Submitting those
>    Submissions.
>    5. *Licenses*.
>
>
>    - *Copyright License*. You grant Microsoft, and those who receive the
>    Submission directly or
>    indirectly from Microsoft, a perpetual, worldwide, non-exclusive,
>    royalty-free, irrevocable license in the
>    Submission to reproduce, prepare derivative works of, publicly
>    display, publicly perform, and distribute
>    the Submission and such derivative works, and to sublicense any or all
>    of the foregoing rights to third
>    parties.
>    - *Patent License*. You grant Microsoft, and those who receive the
>    Submission directly or
>    indirectly from Microsoft, a perpetual, worldwide, non-exclusive,
>    royalty-free, irrevocable license under
>    Your patent claims that are necessarily infringed by the Submission or
>    the combination of the
>    Submission with the Project to which it was Submitted to make, have
>    made, use, offer to sell, sell and
>    import or otherwise dispose of the Submission alone or with the
>    Project.
>    - *Other Rights Reserved*. Each party reserves all rights not
>    expressly granted in this Agreement.
>    No additional licenses or rights whatsoever (including, without
>    limitation, any implied licenses) are
>    granted by implication, exhaustion, estoppel or otherwise.
>
>
>    6. *Representations and Warranties*. You represent that You are
>    legally entitled to grant the above
>    licenses. You represent that each of Your Submissions is entirely Your
>    original work (except as You may
>    have disclosed under Section 3). You represent that You have secured
>    permission from Your employer to
>    make the Submission in cases where Your Submission is made in the
>    course of Your work for Your
>    employer or Your employer has intellectual property rights in Your
>    Submission by contract or applicable
>    law. If You are signing this Agreement on behalf of Your employer, You
>    represent and warrant that You
>    have the necessary authority to bind the listed employer to the
>    obligations contained in this Agreement.
>    You are not expected to provide support for Your Submission, unless
>    You choose to do so. UNLESS
>    REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE
>    WARRANTIES
>    EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED
>    UNDER THIS AGREEMENT IS
>    PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO,
>    ANY WARRANTY OF
>    NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
>    7. *Notice to Microsoft*. You agree to notify Microsoft in writing of
>    any facts or circumstances of which
>    You later become aware that would make Your representations in this
>    Agreement inaccurate in any
>    respect.
>    8. *Information about Submissions*. You agree that contributions to
>    Projects and information about
>    contributions may be maintained indefinitely and disclosed publicly,
>    including Your name and other
>    information that You submit with Your Submission.
>    9. *Governing Law/Jurisdiction*. This Agreement is governed by the
>    laws of the State of Washington, and
>    the parties consent to exclusive jurisdiction and venue in the federal
>    courts sitting in King County,
>    Washington, unless no federal subject matter jurisdiction exists, in
>    which case the parties consent to
>    exclusive jurisdiction and venue in the Superior Court of King County,
>    Washington. The parties waive all
>    defenses of lack of personal jurisdiction and forum non-conveniens.
>    10. *Entire Agreement/Assignment*. This Agreement is the entire
>    agreement between the parties, and
>    supersedes any and all prior agreements, understandings or
>    communications, written or oral, between
>    the parties relating to the subject matter hereof. This Agreement may
>    be assigned by Microsoft.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/PowerPlatformConnectors/pull/3481#issuecomment-2105623663>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABG2JXNUBAMMW4QOXEZ64E3ZBXFNXAVCNFSM6AAAAABHRXDWIOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMBVGYZDGNRWGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

vmanoharas commented on certified-connectors/sendSMS/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,248 @@
+      "post": {
+        "responses": {
+          "default": {
+            "description": "default",
```
'default' response should not be allowed, please correct this.

vmanoharas commented on certified-connectors/sendSMS/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,248 @@
+      "post": {
+        "responses": {
+          "default": {
+            "description": "default",
```
'default' response should not be allowed, please correct this.

vmanoharas commented on certified-connectors/sendSMS/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,248 @@
+{
+  "swagger": "2.0",
```
Kindly update below two properties. 

The 'contact' property is required. ValuePath: /info
The 'x-ms-connector-metadata' property is required.

Please refer the template from the link below. 

https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json

vmanoharas reviewed (commented):
Hello @ElenaMainerici, 

Kindly resolve review comments.

ElenaMainerici committed:
Update Swagger config based on Microsoft review feedback

ElenaMainerici committed:
Merge branch 'microsoft:dev' into dev

ElenaMainerici commented:
Hi @vmanoharas,

I've updated the Swagger configuration for the SendSMS connector based on your feedback. Here's what I changed:
- Updated response schemas for more accurate information.
- Improved visibility settings for parameters in the UI.
- Added necessary security details and updated metadata.

Could you take a look when you have a moment? I appreciate your help and am open to any further suggestions to make sure everything is top-notch.

Thanks!

vmanoharas commented:
> Hi @vmanoharas,
> 
> I've updated the Swagger configuration for the SendSMS connector based on your feedback. Here's what I changed:
> 
> * Updated response schemas for more accurate information.
> * Improved visibility settings for parameters in the UI.
> * Added necessary security details and updated metadata.
> 
> Could you take a look when you have a moment? I appreciate your help and am open to any further suggestions to make sure everything is top-notch.
> 
> Thanks!

Thanks you very much @ElenaMainerici, will review and provide my feedback/approval.

vmanoharas commented on certified-connectors/sendSMS/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,348 @@
+      "propertyValue": "https://www.sendsms.ro/ro/prelucrarea-datelor-cu-caracter-personal-679-2016-gdpr/"
+    },
+    {
+      "propertyName": "Categories",
```
The connector categories should be separated with semicolon (;) not with comma.

vmanoharas reviewed (commented):
Hello @ElenaMainerici, 

Kindly fix one comment.

ElenaMainerici committed:
BugFix

ElenaMainerici committed:
Merge branch 'dev' of github.com:sendSMS-RO/PowerPlatformConnectors into dev

ElenaMainerici commented:
Done :)

vmanoharas reviewed (commented):
Hello @ElenaMainerici, 

Thank you very much for resolving issues. 

Your merge commit has an issue, it has not merged properly, please refer the apiProperties validation checker error. Please recommit or close this PR and open a new PR. Please don't merge it until the review is completed.

![image](https://github.com/microsoft/PowerPlatformConnectors/assets/137378994/ff1e6d4f-7a37-47e3-a627-d538d14056a4)
